itest/tests/zar.test.js

```diff
@@ -0,0 +1,82 @@
+/* @flow */
+
+// This is needed to use zealot outside a browser.
+global.fetch = require("node-fetch")
+
+import {exec, execSync} from "child_process"
+import path from "path"
+
+import zealot from "../../src/js/services/zealot"
+
+import {retryUntil} from "../lib/control"
+import {nodeZqDistDir} from "../lib/env"
+import {handleError, stdTest} from "../lib/jest.js"
+import {
+  click,
+  ingestFile,
+  newAppInstance,
+  reload,
+  startApp,
+  waitForResults
+} from "../lib/app"
+
+describe("Zar tests", () => {
+  let app
+  let testIdx = 0
+  const ZAR = path.join(nodeZqDistDir(), "zar")
+  const ZAR_SPACE_NAME = "sample.zar"
+  beforeEach(() => {
+    app = newAppInstance(path.basename(__filename), ++testIdx)
+    return startApp(app)
+  })
+
+  afterEach(async () => {
+    if (app && app.isRunning()) {
+      return await app.stop()
+    }
+  })
+
+  stdTest(`Brim starts when a Zar space is present`, (done) => {
+    ingestFile(app, "sample.tsv")
+      .then(async () => {
+        await waitForResults(app)
+
+        // Use zealot to
+        // 1. Create a new space
+        // 2. Find the path to sample.tsv.brim's all.zng
+        const client = zealot.client("localhost:9867")
+
+        const sampleSpace = (await client.spaces.list())[0]
+        const zarSpace = await client.spaces.create({name: ZAR_SPACE_NAME})
+
+        const zngFile = path.join(sampleSpace.data_path, "all.zng")
+        const zarRoot = zarSpace.data_path
+
+        // Create a zar archive inside the space and index some
+        // interesting stuff.
+        execSync(`"${ZAR}" import -s 1024B -R "${zarRoot}" "${zngFile}"`)
+        await Promise.all(
+          [":ip", ":port", "uid", "_path"].map((index) =>
+            exec(`"${ZAR}" index -R "${zarRoot}" ${index}`)
```

Comment:
Exec doesn't return a promise by default. It returns a child process object. (which Promise.all will think is a resolved promise. 

You can "promisify it" though. There is an example in the docs here: https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback

```js
const util = require('util');
const exec = util.promisify(require('child_process').exec);

async function lsExample() {
  const { stdout, stderr } = await exec('ls');
  console.log('stdout:', stdout);
  console.error('stderr:', stderr);
}
lsExample();
```

Reply:
I appreciate this feedback. I fixed it in a slightly different way.